src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -798,9 +824,14 @@ protected ArrayBuilder<Symbol> AddOrWrapTupleMembers(ImmutableArray<Symbol> curr
             }
 
             elementsMatchedByFields.Free();
+            members.AddRange(nonFieldMembers);
```

Comment:
>members.AddRange(nonFieldMembers); [](start = 12, length = 34)

Why is it necessary for fields to precede non-fields?

Reply:
test reply

---
In reply to: [597388310](https://github.com/dotnet/roslyn/pull/44231#discussion_r597388310) [](ancestors = 597388310)

Reply:
I had a vague recollection that we may have some code depending on that, but that doesn't seem to be the case. 
I tested the change again just now and only a few tests that verify the members in order are affected.

@AlekseyTs Maybe you remember about this? Is it important to have the fields come first? If not, I'll update the test baseline.

---
In reply to: [597388310](https://github.com/dotnet/roslyn/pull/44231#discussion_r597388310) [](ancestors = 597388310)

Reply:
> Maybe you remember about this? Is it important to have the fields come first? If not, I'll update the test baseline.

I think PENamedTypeSymbol assumes some grouping and ordering of members.

---
In reply to: [601706851](https://github.com/dotnet/roslyn/pull/44231#discussion_r601706851) [](ancestors = 601706851,597388310)